File: app/src/main/java/com/boardgamegeek/util/fabric/CollectionViewManipulationEvent.java
package com.boardgamegeek.util.fabric;

import com.crashlytics.android.answers.Answers;

public class CollectionViewManipulationEvent extends DataManipulationEvent {
	public static void log(String action) {
		CollectionViewManipulationEvent event = new CollectionViewManipulationEvent(action);
		Answers.getInstance().logCustom(event);
	}

	public static void log(String action, String viewName) {
		CollectionViewManipulationEvent event = new CollectionViewManipulationEvent(action);
		event.putCustomAttribute("ViewName", viewName);
		Answers.getInstance().logCustom(event);
	}

	protected CollectionViewManipulationEvent(String action) {
		super("CollectionView");
		putCustomAttribute("action", action);
	}
}

File: app/src/main/java/com/boardgamegeek/util/fabric/CollectionViewManipulationEvent.kt
package com.boardgamegeek.util.fabric

import com.crashlytics.android.answers.Answers

class CollectionViewManipulationEvent(action: String) : DataManipulationEvent("CollectionView") {

    init {
        putCustomAttribute("action", action)
    }

    companion object {
        @JvmStatic
        fun log(action: String) {
            val event = CollectionViewManipulationEvent(action)
            Answers.getInstance().logCustom(event)
        }

        @JvmStatic
        fun log(action: String, viewName: String) {
            val event = CollectionViewManipulationEvent(action).apply {
                putCustomAttribute("ViewName", viewName)
            }
            Answers.getInstance().logCustom(event)
        }
    }
}
